### ISSUE TITLE
Data validation and other tests

### ISSUE THREAD

**Issue by TomHardwicke** [langcog/experimentology#14]
A reviewer suggests we include some advice on data validation. I can add something on this, but my current practice is pretty unsophisticated - I just sprinkle my code with a few `stopifnot()` statements whenever it occurs to me. 

* Does anyone have a more sophisticated workflow? I've tried a few packages like [`validate`](https://cran.r-project.org/web/packages/validate/vignettes/cookbook.html), but they can get quite complex quite quickly and are potentially beyond our scope, or better addressed in an appendix.
* Should we treat data validation and more general testing as separate topics? We could have an appendix on testing that includes both and refer to that from both the data management and reproducible writing chapters?

**Comment by TomHardwicke** [comment]
Yes, something on data validation could fit quite well into the project management chapter. I very briefly mention defensive programming in the writing chapter but without any detail on how to do it practically - is it worth adding a few lines on how to do use code tests in the R Markdown appendix?

**Comment by mcfrank** [comment]
I like the idea of referencing it in the appendix as well. If you didn't
mind adding a bit of prose to each that would be great! Thank you.
